### PR TITLE
Feature/52059 - Select

### PR DESCRIPTION
**Pull request by Aekal**
dev-52059

### CODE REVIEW

File: components/src/molecules/select/Select.html [hunk 1]
```diff
@@ -1,15 +1,14 @@
...
     :class="['select__label', labelClass]"
   >
-    {{ label }}
-   </a-label>
+    <!-- @slot Label text -->
+    <slot />
+  </a-label>
   <div class="select__field">
     <select
-      :id:="id"
```
Comment (szafran89) [inline components/src/molecules/select/Select.html]: we need support for `v-model`. Please consider if we should use scoped slots or there is any other options

File: components/src/molecules/select/Select.stories.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    `,
+    computed: {
+      countryOptions() {
+        return countries.map((country) => {
+          return {
+            value: country.code,
+            text: country.name
+          }
+        })
+      }
```
Comment (szafran89) [inline components/src/molecules/select/Select.stories.js]: move `countryOptions` before stories to reduce duplication

File: components/src/molecules/select/Select.stories.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import {storiesOf} from '@storybook/vue'
```
Comment (szafran89) [inline components/src/molecules/select/Select.stories.js]: checnge to `{ storiesOf }`

File: components/src/molecules/select/Select.stories.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  }))
+  .add('Hidden label', () => ({
+    components: { ASelect },
+    template: `
+      <a-select
+        id="field-id"
+        name="field-name"
+        :options="countryOptions"
+        selected="PL"
+        labelClass="label--hidden"
```
Comment (szafran89) [inline components/src/molecules/select/Select.stories.js]: https://eslint.vuejs.org/rules/attribute-hyphenation.html

File: components/src/molecules/select/Select.stories.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        name="field-name"
+        :options="countryOptions"
+        selected="PL"
+      >
+        Label text
+      </a-select>
+    `,
+    computed: {
+      countryOptions() {
+        return countries.map((country) => {
```
Comment (szafran89) [inline components/src/molecules/select/Select.stories.js]: can we change mock instead of parsing this data?

**Comment by szafran89** [comment]
Additionally we need support for default option, you can also add selected option under rendered component inside stories, sth like here https://bootstrap-vue.js.org/docs/components/form-select

File: components/src/molecules/select/Select.html [hunk 6]
```diff
@@ -29,11 +28,11 @@
...
-        :key="index"
-      >
-        {{ validation.text }}
-      </span>
+    <span
+      v-for="(validation, index) in currentValidations"
+      :key="index"
+    >
+      {{ validation.text }}
+    </span>
```
[the remark this answers is not in the document]
Comment (Aekal) [reply components/src/molecules/select/Select.html]: ok, removed

File: components/src/molecules/select/Select.html (hunk 1 above)
Comment (Aekal) [reply components/src/molecules/select/Select.html]: support for v-model added, you can see changing value in actions tab, it now works similar to vsf `BaseSelect`

File: components/src/molecules/select/Select.stories.js (hunk 2 above)
Comment (Aekal) [reply components/src/molecules/select/Select.stories.js]: removed, using mock instead

File: components/src/molecules/select/Select.stories.js (hunk 3 above)
Comment (Aekal) [reply components/src/molecules/select/Select.stories.js]: done

File: components/src/molecules/select/Select.stories.js (hunk 4 above)
Comment (Aekal) [reply components/src/molecules/select/Select.stories.js]: I always forgot about it, that's why we need eslint :D

File: components/src/molecules/select/Select.stories.js (hunk 5 above)
Comment (Aekal) [reply components/src/molecules/select/Select.stories.js]: done

File: components/src/molecules/select/Select.html (hunk 1 above)
Comment (Aekal) [reply components/src/molecules/select/Select.html]: I'm not sure if it's better to show value in actions tab or directly under select component as we have in `swatch` component, what do you prefer?

**Comment by Aekal** [comment]
Ok, I've finally got it working with `v-model` and `@change` event. Variant with unselected option added

File: components/src/molecules/select/Select.html (hunk 1 above)
Comment (szafran89) [reply components/src/molecules/select/Select.html]: actions addon is ok.

File: components/src/molecules/select/Select.html [hunk 7]
```diff
@@ -1,39 +1,32 @@
 <div class="select">
   <a-label
     :for="id"
-    :hidden="hiddenLabel"
-    :inline="inlineLabel"
     :class="['select__label', labelClass]"
   >
-    {{ label }}
-   </a-label>
+    <!-- @slot Label text -->
```
Comment (szafran89) [inline components/src/molecules/select/Select.html]: Maybe better `Label content`?

File: components/src/molecules/select/Select.html [hunk 8]
```diff
@@ -1,39 +1,32 @@
...
       @change="$emit('input', $event.target.value)"
     >
-      <option v-if="!selected" />
+      <option disabled selected v-if="!value">
+        <!-- @slot Default option text -->
+        <slot name="option"/>
+      </option>
       <option
         v-for="(option, key) in options"
         :key="key"
```
Comment (szafran89) [inline components/src/molecules/select/Select.html]: use `option.value` and remove key from `v-for`

File: components/src/molecules/select/Select.stories.js [hunk 9]
```diff
@@ -0,0 +1,89 @@
+import { storiesOf } from '@storybook/vue'
+import { action } from '@storybook/addon-actions'
+
+import countries from '../../../mocks/countries.json'
+
+import ASelect from './Select.vue'
+
+storiesOf('Molecules/Select', module)
+  .addParameters({ info: true })
+  .add('Default', () => ({
```
Comment (szafran89) [inline components/src/molecules/select/Select.stories.js]: Please format the same way as radio/checkbox

Comment (Aekal) [reply components/src/molecules/select/Select.stories.js]: I'm not a huge fan of this formatting, but ok, done :P

File: components/src/molecules/select/Select.stories.js (hunk 4 above)
Comment (szafran89) [reply components/src/molecules/select/Select.stories.js]: Yes, but IMO will not work inside story

File: components/src/molecules/select/Select.stories.js [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+          id="field-id"
+          name="field-name"
+          :options="countries"
+          v-model="selectedCountry"
+          @input="showCountry"
+        >
+          <template v-slot:label>
+            Label text
+          </template>
+          <template v-slot:option>
```
Comment (szafran89) [inline components/src/molecules/select/Select.stories.js]: please use shortcut from vue 2.6.0 in all places, then can go to develop

Comment (Aekal) [reply components/src/molecules/select/Select.stories.js]: Done

**Pull request merged**